examples/source/1.components/amp-web-push/static/sw.js

```diff
@@ -256,33 +166,106 @@ function broadcastReply(command, payload) {
       });
 }
 
+
+/**
+  This section of the ServiceWorker deals with SW events: initializing the DB where tokens will be persisted,
+  sending the request for push notifcations to the endpoint, and listening to push events.
+ */
+
+/**
+  Using skipWaiting() to force the waiting service worker to become the active service worker.
+ */
+self.addEventListener('install', () => self.skipWaiting());
+
+/**
+  Creates the DB to store subscription objects and calls clients.claim(), to allow an active
+  service worker to set itself as the controller for all clients within its scope.
+ */
+self.addEventListener('activate', (event) => {
+  event.waitUntil((async () => {
+    await idb.open(WEB_PUSH_DB, 1, (upgradeDB) => {
+      upgradeDB.createObjectStore(WEB_PUSH_SUBSCRIPTION, {
+        keyPath: 'id',
+      });
+    });
+    self.clients.claim();
+  })());
+});
+
+/**
+  Listens to push events, and displays a notification, using the payload text.
+ */
+self.addEventListener('push', (event) => {
+  const options = {
+    body: event.data.text(),
+    vibrate: [100, 50, 100],
+    data: {
+      dateOfArrival: Date.now(),
+      primaryKey: 1,
+    },
+  };
+  event.waitUntil(
+      self.registration.showNotification('Push Notification', options)
+  );
+});
+
+/**
+  Intercepts the /send-push request: retrieves the subscription object from indexedDB, and passes it to an endpoint
+  so WebPush messages can be sent.
+  Any other request goes to the network directly.
+ */
+self.addEventListener('fetch', (event) => {
+  if (event.request.url.includes('/documentation/examples/components/amp-web-push/send-push')) {
+    event.respondWith((async () => {
+      const db = await idb.open(WEB_PUSH_DB, 1);
+      const tx = db.transaction([WEB_PUSH_SUBSCRIPTION], 'readonly');
+      const store = tx.objectStore(WEB_PUSH_SUBSCRIPTION);
+
+      const subscriptionJSON = await store.get(WEB_PUSH_SUBSCRIPTION_ID);
+      const options = {
+        method: 'POST',
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: subscriptionJSON.data,
+      };
+
+      return await fetch(self.location.origin +
```

Comment:
The `await` is not needed here. It's fine to return a promise.

Reply:
PTAL